Enable to run TPCDS performance benchmarks on EKS

grzegorz8 opened the pull request:
## Description

This PR is a follow-up of https://github.com/delta-io/delta/pull/1179, which added terraform for benchmarks infrastructure deployment. This PR, in turn, enables to run TCPDS performance benchmarks on AWS EKS.

Please see the `benchmarks/infrastructure/aws/terraform/README.md` updates for more details.

## How was this patch tested?

I manually ran the benchmarks on AWS EKS.

What is more, I ran benchmarks on AWS EMR and Google Dataproc to ensure nothing is broken.

## Does this PR introduce _any_ user-facing changes?

No.

Commit:
Enable to run TPCDS performance benchmarks on EKS

tdas commented on benchmarks/infrastructure/aws/terraform/README.md:
```diff
@@ -1,4 +1,11 @@
+The document describes step by step how to run Delta Lake benchmarks on AWS infrastructure.
+Currently, you can run benchmarks on either EMR on EKS.
+
+## Prerequisites
```
I think a big question here thats not obvious to me is how do you configure what version of spark will be installed?

grzegorz8 replied:
For EKS I can add a "spark-version" parameter, so that a proper spark image will be used.
For EMR I can add yet another terraform variable for EMR version, so the user can indirectly choose which spark version to use.

Of course I'll describe it in README. Is that ok?

Commit:
Allow to merge BenchmarkSpecs

Commit:
Update AWS permissions list

Commit:
Allow to specify spark version in EKS benchmarks

Commit:
Extend README

Commit:
Add emr_version terraform variable

emr_version parameter allows to indirectly control Spark version.

Commit:
Fix Kubectl command runner

Commit:
Use EMR-6.6.0 be default

EMR-6.6.0 contains Spark 3.2.0 which is compatible with Delta 2.0.0.

Commit:
Add common random suffix to each resource name

Adding a random suffix allow to run benchmarks in parallel in the
same account/project.

Commit:
Remove unused kubernetes provider

dennyglee reviewed (approved):
LGTM!

foss-contributor commented:
@grzegorz8, thank you for your contribution! This pull request has been inactive for over 18 months and will be closed in one week. To keep it open, please leave a comment on this pull request or contact us in the `#contributors` channel on [Delta Lake Slack](https://go.delta.io/slack). We appreciate your involvement in Delta Lake!
